ReactCommon/fabric/components/text/paragraph/ParagraphShadowNode.cpp

```diff
@@ -38,21 +38,20 @@ void ParagraphShadowNode::setMeasureCache(
   measureCache_ = cache;
 }
 
-void ParagraphShadowNode::updateLocalDataIfNeeded() {
+void ParagraphShadowNode::updateStateIfNeeded() {
   ensureUnsealed();
 
   auto attributedString = getAttributedString();
-  auto currentLocalData =
-      std::static_pointer_cast<const ParagraphLocalData>(getLocalData());
-  if (currentLocalData &&
-      currentLocalData->getAttributedString() == attributedString) {
+  auto const &state = getStateData();
+  if (state.attributedString == attributedString) {
     return;
   }
 
-  auto localData = std::make_shared<ParagraphLocalData>();
-  localData->setAttributedString(std::move(attributedString));
-  localData->setTextLayoutManager(textLayoutManager_);
-  setLocalData(localData);
+  auto newStateData = ParagraphState();
```

Comment:
`setStateData({attributedString, textLayoutManager_});`

Reply:
Also, I admit the confusion between State and StateData (names of variables and names of classes). Now it's inconsistent.

I am thinking that we should rename `State` to `StateContainer` (or `StateHolder`), and use `State` as `StateData`. This way we can improve product/component development experience.

That why some `*StateData` classes are named with `*State` suffix (e.g. `ParagraphState` here).

Reply:
gotcha, I would agree with that too. will make the change to be more succinct too :)